### DISCUSSION TITLE
Angular Routes

### DISCUSSION THREAD

**Discussion by schoes** [fastify/fastify#5119] [Q&A] [answered]
Hello there
I got a question my app uses the @fastify/static plugin and serves an angular app under the following config : 
```
fastify.register(fastifyStatic, {
    root: clientDir,
    prefix: `/app`,
    prefixAvoidTrailingSlash: true,
    maxAge: SEVEN_DAYS,
    cacheControl: true,
    etag: true,
    lastModified: true
});
```
When i navigate directly to /app everything works and the angular apps bootstraps without any problem. But when I navigate to /app/route1 which is a known route in the angular app i get a 404 from fastify. How can I achieve this that the angular routes get routed in fastify ? THX

**Comment by L2jLiga** [comment, marked as answer]
I guess you have to use `setNotFoundHandler` in order to return index.html on unknown routes

https://angular.io/guide/deployment#routed-apps-must-fall-back-to-indexhtml

```typescript
import { readFile } from 'node:fs/promises';
import { join } from 'node:path';

app.setNotFoundHandler((request) => readFile(join(clientDir, 'index.html')));
```

**Reply by schoes** [reply]
Will try it and get back with the result . THX

**Reply by schoes** [reply]
when i use the upper code snippet and read the index.html chrome downloads the file instead of rendering it. what do i have do change in order to load the file instead of downloading it ? . i also tried with reply.sendFile but that doesn't work

**Reply by L2jLiga** [reply]
What about something like this?

```typescript
app.setNotFoundHandler(async (request, reply) => {
  const content = await readFile(join(clientDir, 'index.html'), 'utf8')
  reply.type('text/html').status(200);
  return content;
});
```

**Reply by amit78523** [reply]
I have prefix as '/' then i am getting, setNotFoundHandler is already set for '/'

**Comment by schoes** [comment]
When i configure the NotFoundHandler like: 
`fastify.setNotFoundHandler((request, reply) => {
    reply.header('Content-Type', 'text/html');
    reply.send(createReadStream(join(clientDir, 'index.html'), 'utf-8'));
});`
the application loads the index.html instead of downloading it
